sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2835,6 +2835,84 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
     checkAnswer(df, Row(1, 3, 4) :: Row(2, 3, 4) :: Row(3, 3, 4) :: Nil)
   }
 
+  test("Support filter clause for aggregate function with hash aggregate") {
+    Seq(("APPROX_COUNT_DISTINCT(a)", 3), ("COUNT(a)", 3), ("FIRST(a)", 1), ("LAST(a)", 3),
```

Comment:
The thing we want to test is, FILTER works in hash aggregate. We don't need to test a lot of different agg functions and prove Spark use hash aggregate to execute them.

Let's just test one function that uses `HashAggregateExec`, and one function that uses `ObjectHashAggregateExec `

Reply:
OK. I will only keep one function in each case.